storage/aptosdb/src/pruner/worker.rs

```diff
@@ -48,27 +51,22 @@ impl Worker {
         while self.receive_commands() {
             // Process a reasonably small batch of work before trying to receive commands again,
             // in case `Command::Quit` is received (that's when we should quit.)
-            let mut error_in_pruning = false;
-            let mut ledger_db_batch = SchemaBatch::new();
-            for db_pruner in &self.db_pruners {
-                let result = db_pruner
-                    .lock()
-                    .prune(&mut ledger_db_batch, self.max_version_to_prune_per_batch);
-                result.map_err(|_| error_in_pruning = true).ok();
-            }
-            // Commit all the changes to DB atomically
-            self.ledger_db
-                .write_schemas(ledger_db_batch)
-                .map_err(|_| error_in_pruning = true)
-                .ok();
+            let error_in_pruning = AtomicBool::new(false);
+            let max_version_to_pruner = self.max_version_to_prune_per_batch;
+            self.db_pruners.par_iter().for_each(|db_pruner| {
```

Comment:
Not a good idea to do IOs in the global rayon thread pool. At the least we should use our own pool (of two threads). Better: can we abstract the common logic out and initialize the ledger and state pruner workers on their own threads?

Reply:
We might also want to support a mode to take a threadpool/executor as an input, if we want to support fine grained tuning in the future.